paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RemoveOrphanFilesAction.java

```diff
@@ -0,0 +1,522 @@
+                return ReadingResult.forException(readingResult.originalException());
+            }
+            readingResult.result().stream()
+                    .map(ManifestEntry::file)
+                    .forEach(
+                            f -> {
+                                dataFiles.add(f.fileName());
+                                dataFiles.addAll(f.extraFiles());
+                            });
+        }
+        return ReadingResult.forResult(dataFiles);
+    }
+
+    /**
+     * If the snapshot/tag which are being read still exists, throw exception to stop removing
+     * process.
+     */
+    private void checkSnapshotAndTag(
+            Snapshot currentSnapshot,
+            SnapshotManager snapshotManager,
+            TagManager tagManager,
+            SortedMap<Snapshot, String> tags,
+            @Nullable Exception optionalException)
+            throws IOException {
+        long id = currentSnapshot.id();
+        if (snapshotManager.snapshotExists(id)) {
+            throw new IOException(
+                    String.format(
+                            "Failed to read snapshot %s when trying to find used files. "
+                                    + "To avoid deleting used files, stop removing process.",
+                            id),
+                    optionalException);
+        } else {
+            String tagName = tags.get(currentSnapshot);
+            if (tagName != null && tagManager.tagExists(tagName)) {
+                throw new IOException(
+                        String.format(
+                                "Failed to read tag %s when trying to find used files. "
+                                        + "To avoid deleting used files, stop removing process.",
+                                tagName),
+                        optionalException);
+            }
+        }
+    }
+
+    /**
+     * List all the Paimon files, filter them by olderThanMillis by snapshots and tags, and store
+     * them in view ALL(path STRING, fileName STRING).
+     */
+    private void buildAllFilesView(FileStoreTable table) throws Exception {
+        RowTypeInfo rowTypeInfo =
+                new RowTypeInfo(
+                        new TypeInformation[] {
+                            BasicTypeInfo.STRING_TYPE_INFO, BasicTypeInfo.STRING_TYPE_INFO
+                        },
+                        new String[] {"path", "fileName"});
+
+        DataStream<Row> allFilesDataStream =
+                env.fromCollection(
+                                listPaimonFileDirs(
+                                        table.fileIO(),
+                                        table.location(),
+                                        table.partitionKeys().size()))
+                        .flatMap(new ListCandidates(table.fileIO(), olderThanMillis))
+                        .returns(rowTypeInfo);
+        batchTEnv.createTemporaryView("ALL", allFilesDataStream);
```

Comment:
Can we just use `COMMON_IO_FORK_JOIN_POOL`? We don't need to use Flink Datastream